aiohttp_client_cache/response.py

```diff
@@ -91,7 +91,10 @@ async def from_client_response(
         response._body = client_response._body
         client_response.content = CachedStreamReader(client_response._body)
 
-        # Set remaining attributes individually
+        # Set remaining attributes individually.
+        if expires is not None:
+            # An unrecoverable error (wrong library low-level API usage).
+            assert expires.tzinfo is None, f'Expected a naive datetime, but got {expires}.'
```

Comment:
`TypeError` would be more appropriate here than `AssertionError`:
```py
if expires is not None and expires.tzinfo is not None:
    raise TypeError(f'Expected a naive datetime, but got {expires}.')
```

Reply:
Poorly written user-side code can look like this 

```py
try:
    # some user's unrelated code here...
    session.get(url)
    # some user's unrelated code here...
except TypeError:
    pass
```

and libraries must raise custom errors. 

This guarantees that the error in not confused with something else. 

How about defining `class UnsupportedExpiresError(Exception)`?